libraries/AP_Motors/AP_MotorsHeli_Dual.cpp

```diff
@@ -265,6 +271,10 @@ void AP_MotorsHeli_Dual::calculate_armed_scalars()
     _rotor.set_critical_speed(_rsc_critical*0.001f);
     _rotor.set_idle_output(_rsc_idle_output*0.001f);
     _rotor.set_throttle_curve(thrcrv, (uint16_t)_rsc_slewrate.get());
+    _rotor.set_governor_disengage(_rsc_governor_disengage*0.01f);
+    _rotor.set_governor_droop_setting(_rsc_governor_droop_setting*0.01f);
+    _rotor.set_governor_setpoint(_rsc_governor_setpoint);
```

Comment:
The setpoint for the RSC is a PWM value for ESC's. The governor setpoint uses actual rotor rpm. Since they are two essentially different things I made them separate.

Reply:
All these estimated speeds, including critical speed, are not governor values - they are pwm values and governors deal with rpm. That is one of the reasons why I added the _governor_status variable that switches from 0 to 1 when the governor is at full operational output, meaning the difference between the reference and setpoint is within the operational droop limit of the governor.

None of the principles of operation of a droop speed control apply to, nor can use, any of the RSC stuff except for the throttle curve. Because the rest of it all based on pwm values for throttle.

Droop speed controls have three components:
1) Setpoint
2) Reference Speed
3) Fuel control
In ArduPilot the Setpoint is the desired governor speed. The reference speed is the measured rpm value. The fuel control is throttle output. Droop is the difference between setpoint and reference speed. Reference speed goes down with increasing load, and this is what provides the stability of a droop speed control. Fuel control is proportional to droop, either linear, sqrt, or x^2 depending on the governor design.

The ramp time is still used to ramp the throttle, the runup time is still used to hold things until the _governor_status goes to 1. But just like any of the other modes, the ramp time must be determined by the end user because not all engines and rotor systems accelerate at the same rate. That was the primary reason a timer does not work to soft-start a droop governor. A percentage of the fuel control works the best for soft-starting, since fuel control is directly related to the amount of torque applied vs throttle opening. Maximum torque is achieved at WOT and peak BMEP, etc.. And the power (horsepower) relationship is directly linked to torque and rpm over time (hp = (torque * rpm) / 5252. So using this principle of how much fuel control is applied predicts the acceleration speed quite accurately for all engine types, based on a properly tuned throttle curve, that fits the prime mover's torque curve.

So in a nutshell that is the basics of how a droop speed control works. How to integrate that with the existing stuff in the RSC for other modes, I'm at a loss. Most of the rest of it was designed for electrics. This is the first ever fuel engine control mode in Copter.

Reply:
I’m just saying it doesn’t matter what the variable represents PWM or rpm. As long as the variable is of the same type then they can be interchangeable. And it is not like your using two RSC modes at the same time.

Reply:
So you're saying we could use current H_RSC_SETPOINT set to PWM value for electric ESC, but change it to rpm value for Mode 4 governor? In that case, what do we use for a default?

Reply:
Ok, that will work. I'm working on rebasing this and re-ordering the commits so it can be more easily pushed to for further changes. As soon as I get that done I'll do a force push to my branch and it should be cleaned up. I left the commits to Copter individual, but look it over after I push it, I could squash those Copter commits too. Unless there is an issue with the call in Heli.cpp for the rpm, which will now be squashed into the AP_Motors. We may want to move that call to the RSC?